oauth2l-go/src/oauth2client/oauth2client.go

```diff
@@ -0,0 +1,298 @@
+package oauth2client
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	// The URN for getting verification token offline
+	oobCallbackUrn = "urn:ietf:wg:oauth:2.0:oob"
+	// The URN for token request grant type jwt-bearer
+	jwtBearerUrn = "urn:ietf:params:oauth:grant-type:jwt-bearer"
+	// Common prefix for google oauth scope
+	scopePrefix = "https://www.googleapis.com/auth/"
```

Comment:
Move this line to oauth2l.go. This entire file should have zero dependency on Google.